src/main/java/org/alfresco/heartbeat/InfoDataCollector.java

```diff
@@ -126,7 +138,43 @@ public void afterPropertiesSet() throws Exception
         infoValues.put("schema", new Integer(serverDescriptor.getSchema()));
         infoValues.put("edition", serverDescriptor.getEdition());
         infoValues.put("deploymentMethod", deploymentMethodProvider.getDeploymentMethod().toString());
+        infoValues.put("os.vendor",System.getProperty("os.name"));
+        infoValues.put("os.version",System.getProperty("os.version"));
+        infoValues.put("os.arch",System.getProperty("os.arch"));
 
+        infoValues.put("java.vendor",System.getProperty("java.vendor"));
+        infoValues.put("java.version",System.getProperty("java.version"));
+        
+
+        Connection con = null;
+        try
+        {
+            con = dataSource.getConnection();
+            DatabaseMetaData dbmeta = con.getMetaData();
+
+            infoValues.put("db.vendor", dbmeta.getDatabaseProductName());
+            infoValues.put("db.version", dbmeta.getDatabaseProductVersion());
+            infoValues.put("db.driverName", dbmeta.getDriverName());
+            infoValues.put("db.driverVersion", dbmeta.getDriverVersion());
+//            infoValues.put("db driverClass", dbmeta.getDriver); IS THERE ANY ? 
+        } 
+        catch (SQLException e)
+        { //TODO:            
```

Comment:
Do not leave a TODO here.  Is the exception worth logging or just a single-line WARN that the values could not be retrieved.

Reply:
In some of our other collectors, if we can't get a value for something, we just don't add it without logging anything. LicenseDataCollector is an example of this.

Reply:
Then remove the TODO and put a comment to that effect, please.